packages/amplify_datastore/example/lib/main.dart

```diff
@@ -92,7 +116,8 @@ class _MyAppState extends State<MyApp> {
     });
 
     (await Amplify.DataStore.query(Post.classType,
-            where: Post.ID.ne("e25859fc-e254-4e8b-8cae-62ccacce4097")))
+            where: QueryField(fieldName: "post.id")
```

Comment:
@Amplifiyer this is an open issue on Android and iOS, as well. It is mentioned [in the public documentation, here](https://docs.amplify.aws/lib/datastore/data-access/q/platform/android#predicates):

> Note: when constructing predicates, static `QueryField` instances such as `Post.RATING` do not own any information about the model to which the field belongs. In order to avoid any ambiguity between field names which are used across multiple models, it is recommended to construct a custom instance of `QueryField` in the form of `QueryField.field("{model-name}.{field-name}")` (i.e. `field("post.rating")`).

Please see https://github.com/aws-amplify/amplify-android/issues/908 for some additional discussion and a possible resolution.

TLDR: it's an easy bug to fix, by updating your Dart codegen. The generated `QueryField`s in your model should say `ModelName_id` instead of just `id`. Android and iOS still need this fix.

Reply:
Thanks Jameson, I believe its all the more reasons for us to fix it right away since we are in the process of implementing the codegen.

Reply:
Will speak with codegen team about fixing this.  In the meantime see the changes I made to the codegen models to use `modelname.id` instead of just `id`